pkg/providers/vsphere/reconciler/reconciler.go

```diff
@@ -131,15 +131,21 @@ func (r *Reconciler) ReconcileWorkerNodes(ctx context.Context, log logr.Logger,
 	).Run(ctx, log, clusterSpec)
 }
 
-// ValidateControlPlaneIP validates that the control plane IP is unique.
+// ValidateControlPlaneIP validates that the control plane IP is unique only on cluster creation.
 func (r *Reconciler) ValidateControlPlaneIP(ctx context.Context, log logr.Logger, clusterSpec *c.Spec) (controller.Result, error) {
 	log = log.WithValues("phase", "validateControlPlaneIP")
 
-	if err := r.ipValidator.ValidateControlPlaneIPUniqueness(clusterSpec.Cluster); err != nil {
-		clusterSpec.Cluster.Status.FailureMessage = ptr.String(err.Error())
-		log.Error(err, "Unavailable control plane IP")
-		return controller.ResultWithReturn(), nil
+	cluster := &anywherev1.Cluster{}
+	clusterName := types.NamespacedName{Namespace: clusterSpec.Cluster.ObjectMeta.Namespace, Name: cluster.Name}
+
+	if err := r.client.Get(ctx, clusterName, cluster); err != nil && apierrors.IsNotFound(err) {
+		if err = r.ipValidator.ValidateControlPlaneIPUniqueness(clusterSpec.Cluster); err != nil {
+			clusterSpec.Cluster.Status.FailureMessage = ptr.String(err.Error())
+			log.Error(err, "Unavailable control plane IP")
+			return controller.ResultWithReturn(), nil
+		}
```

Comment:
You are missing the actual error case handling here if the api error is different from "NotFound"